airflow/providers/samba/hooks/samba.py

```diff
@@ -245,10 +245,22 @@ def setxattr(self, path, attribute, value, flags=0, follow_symlinks=True):
             **self._conn_kwargs,
         )
 
-    def push_from_local(self, destination_filepath: str, local_filepath: str):
-        """Push local file to samba server."""
-        with open(local_filepath, "rb") as f, self.open_file(destination_filepath, mode="wb") as g:
-            copyfileobj(f, g)
+    def push_from_local(self, destination_filepath: str, local_filepath: str, buffer_len: int | None = None):
+        """
+        Push local file to samba server.
+
+        :param destination_filepath: the samba location to push to
+        :param local_filepath: the file to push
+        :param buffer_len:
+            size in bytes of the individual chunks of file to send. Larger values may
+            speed up large file transfers
+        """
+        if buffer_len is not None:
+            with open(local_filepath, "rb") as f, self.open_file(destination_filepath, mode="wb") as g:
+                copyfileobj(f, g, buffer_len)
+        else:  # Use default buffer size for OS as determined by copyfileobj
+            with open(local_filepath, "rb") as f, self.open_file(destination_filepath, mode="wb") as g:
+                copyfileobj(f, g)
```

Comment:
[`shutil.copyfileobj`](https://docs.python.org/3/library/shutil.html#shutil.copyfileobj) always use buffer if length >= 0 (default 0) in case of 0 it fallback to platform specific valuem something like 16-64K for Linux.

So I guess there is no need to create a condition if value provided or not, and just use
`copyfileobj(f, g, buffer_len or 0)`

Reply:
or if you want to keep conditional, e.g. prevent some changes in the future into the https://github.com/python/cpython/blob/main/Lib/shutil.py, you might make arguments by condition, e.g.

```python
args = (f, g, buffer_len) if buffer_len else (f, g)
with open(local_filepath, "rb") as f, self.open_file(destination_filepath, mode="wb") as g:
    copyfileobj(*args)
```

Reply:
Good idea, I'll go with that first one.

Reply:
Actually, the second option made it so that we don't have to change a bunch of the old tests